front/src/applications/operationalStudies/views/v2/SimulationResultsV2.tsx

```diff
@@ -1,276 +1,278 @@
-//   const [initialHeightOfSpaceCurvesSlopesChart, setInitialHeightOfSpaceCurvesSlopesChart] =
-//     useState(heightOfSpaceCurvesSlopesChart);
+  const [timeScaleDomain, setTimeScaleDomain] = useState<TimeScaleDomain>({
+    range: undefined,
+    source: undefined,
+  });
 
-//   const [timeScaleDomain, setTimeScaleDomain] = useState<TimeScaleDomain>({
-//     range: undefined,
-//     source: undefined,
-//   });
+  // X scale domain shared between SpeedSpace and SpaceCurvesSlopes charts.
+  const [positionScaleDomain, setPositionScaleDomain] = useState<PositionScaleDomain>({
+    initial: [],
+    current: [],
+    source: undefined,
+  });
 
-//   // X scale domain shared between SpeedSpace and SpaceCurvesSlopes charts.
-//   const [positionScaleDomain, setPositionScaleDomain] = useState<PositionScaleDomain>({
-//     initial: [],
-//     current: [],
-//     source: undefined,
-//   });
+  const {
+    selectedTrain,
+    selectedTrainRollingStock,
+    selectedTrainPowerRestrictions,
+    trainSimulation,
+    pathProperties,
+  } = useSimulationResults();
 
-//   const {
-//     allowancesSettings,
-//     selectedTrain,
-//     selectedProjection,
-//     simulation,
-//     simulationIsPlaying,
-//     dispatchUpdateSelectedTrainId,
-//     dispatchPersistentUpdateSimulation,
-//   } = useStoreDataForSpaceTimeChart();
+  const handleKey = (e: KeyboardEvent) => {
+    if (e.key === 'z' && e.metaKey) {
+      dispatch(persistentUndoSimulation());
+    }
+    if (e.key === 'e' && e.metaKey) {
+      dispatch(persistentRedoSimulation());
+    }
+  };
 
-//   const { data: selectedTrainSchedule } = osrdEditoastApi.endpoints.getTrainScheduleById.useQuery(
-//     {
-//       id: selectedTrain?.id as number,
-//     },
-//     { skip: !selectedTrain }
-//   );
+  useEffect(() => {
+    // Setup the listener to undi /redo
+    window.addEventListener('keydown', handleKey);
+    return function cleanup() {
+      window.removeEventListener('keydown', handleKey);
+      dispatch(updateTrainIdUsedForProjection(undefined));
+    };
+  }, []);
 
-//   const { data: selectedTrainRollingStock } =
-//     osrdEditoastApi.endpoints.getLightRollingStockByRollingStockId.useQuery(
-//       {
-//         rollingStockId: selectedTrainSchedule?.rolling_stock_id as number,
-//       },
-//       { skip: !selectedTrainSchedule }
-//     );
+  useEffect(() => {
+    if (extViewport !== undefined) {
+      dispatch(
+        updateViewport({
+          ...extViewport,
+        })
+      );
+    }
+  }, [extViewport]);
 
-//   const handleKey = (e: KeyboardEvent) => {
-//     if (e.key === 'z' && e.metaKey) {
-//       dispatch(persistentUndoSimulation());
-//     }
-//     if (e.key === 'e' && e.metaKey) {
-//       dispatch(persistentRedoSimulation());
-//     }
-//   };
+  useEffect(() => {
+    if (trainSimulation && trainSimulation.status === 'success') {
+      const { positions } = trainSimulation.base;
```

Comment:
:question: Why do you use base and not final_output here ?
```suggestion
      const { positions } = trainSimulation.final_output;
```

Reply:
Done